timber-junit/src/main/java/net/lachlanmckee/timberjunit/TimberTestRule.java

```diff
@@ -226,10 +229,12 @@ protected void log(int priority, String tag, String message, Throwable t) {
          * Flushes all the previously stored log messages.
          */
         private void flushLogs() {
-            Iterator<String> iterator = mLogMessageBuffer.iterator();
-            while (iterator.hasNext()) {
-                System.out.println(iterator.next());
-                iterator.remove();
+            synchronized (bufferLock) {
+                Iterator<String> iterator = mLogMessageBuffer.iterator();
+                while (iterator.hasNext()) {
+                    System.out.println(iterator.next());
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Replace this use of System.out or System.err by a logger. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS106)